samples/matter/closure/src/app_task.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "app_task.h"
+
+#include "ClosureControlEndpoint.h"
+#include "ClosureManager.h"
+#include "app/matter_init.h"
+#include "app/task_executor.h"
+#include "board/board.h"
+#include "lib/core/CHIPError.h"
+#include "lib/support/CodeUtils.h"
+
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/data-model/Nullable.h>
+#include <app/util/attribute-storage.h>
+#include <setup_payload/OnboardingCodesUtil.h>
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
+
+using namespace ::chip;
+using namespace ::chip::app;
+using namespace ::chip::DeviceLayer;
+using namespace chip::app::Clusters::Descriptor;
+
+static const struct pwm_dt_spec sPhysicalIndicatorPwmDevice = PWM_DT_SPEC_GET(DT_ALIAS(pwm_led1));
+
+AppTask &AppTask::Instance()
```

Comment:
Instance(), constructor and kClosureEndpoint assignment could be moved to app_task.h file.